src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -104,10 +107,15 @@ class PeerManagerActor(
       stash()
 
     case (HandlePeerConnection(connection, remoteAddress), Success(_)) =>
-      context.unbecome()
-      val peer = createPeer(remoteAddress)
-      peer.ref ! PeerActor.HandleConnection(connection, remoteAddress)
-      unstashAll()
+      if (!isConnectionHandled(remoteAddress)) {
+        context.unbecome()
+        val peer = createPeer(remoteAddress)
+        peer.ref ! PeerActor.HandleConnection(connection, remoteAddress)
+        unstashAll()
+      } else {
+        log.info("Another connection with {} is already opened. Disconnecting.", remoteAddress)
+        connection ! PoisonPill
+      }
 
     case (HandlePeerConnection(connection, remoteAddress), Failure(_)) =>
```

Comment:
In this case the peer limit has been reached (note, the `Failure` in the tuple). The peer will be disconnected and closed anyway (see `       peer.ref ! PeerActor.DisconnectPeer(Disconnect.Reasons.TooManyPeers)` few lines below)